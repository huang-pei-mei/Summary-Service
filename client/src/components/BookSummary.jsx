import React from 'react';
import Summary from './Summary.jsx';



const BookSummary = (props) => {


  return (
    <div className="summary">
      <h2 className="summary-heading">Publisher's Summary</h2>
      <div className="publisher-summary">
        <span className="summary-detail">{props.summary.summary.split('\n').map((str,index)  => <p key={index}>{str}</p>)}</span>
        <span className="copyright">{props.summary.copyright}</span>
      </div>
    </div>
  )

};

export default BookSummary;


