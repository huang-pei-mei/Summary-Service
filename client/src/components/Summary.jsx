import React from 'react';
import BookSummary from './BookSummary.jsx';

const Summary = (props) => (

 <div>
   {/* {console.log(props)} */}

  {
     props.summaries.map((summary, index)=>{
     return (
     < BookSummary summary={summary} key={index}/>
     )
    })
  }

  </div>
  );

export default Summary;




