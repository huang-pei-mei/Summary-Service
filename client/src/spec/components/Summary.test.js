import React from 'react';
import { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Summary from '../../components/Summary.jsx'
import BookSummary from '../../components/BookSummary.jsx'
configure({ adapter: new Adapter() });

describe('Summary', () => {
  it("should render BookSummary", () => {
  const wrapper = shallow(<Summary summaries={['summary']} />);
  const summary = wrapper.find(BookSummary);
  expect(summary.exists()).toBe(true);
});

it('Summary renders div', () => {
  const wrapper = shallow(<Summary summaries={['summary']} />);
  expect(wrapper.find('div').length).toEqual(1);
});

// it('displays summaries if passed props', () => {
//   const wrapper = shallow(<Summary summaries={['summary']} />);
//   expect(wrapper.find('BookSummary').exists()).toEqual(true);
// });

it('should render the state', () => {
  const wrapper = shallow(<Summary summaries={['summary']} />)
  const result = wrapper.find('div').text();
  expect(result).toEqual("<BookSummary />");

});
});


