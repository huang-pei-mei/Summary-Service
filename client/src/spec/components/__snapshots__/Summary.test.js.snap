// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary component should render the state correctly 1`] = `ShallowWrapper {}`;
